pkg/util/nativeutils/eventbus/listener.go

```diff
@@ -125,21 +132,17 @@ func newMultiListener() *multiListener {
 	}
 }
 
-func (m *multiListener) Notify(topic topics.Topic, r bytes.Buffer) {
-	if m.Has([]byte{byte(topic)}) {
-		// creating a new Buffer carrying also the topic
-		tpcMsg := topic.ToBuffer()
-		if _, err := tpcMsg.ReadFrom(&r); err != nil {
-			log.WithField("topic", topic.String()).WithError(err).Warnln("error in writing topic to a multi-dispatched packet")
-			return
-		}
+func (m *multiListener) Forward(topic topics.Topic, msg message.Message) {
+	if !m.Has([]byte{byte(topic)}) {
+		// TODO: maybe log this
```

Comment:
Not sure if that's necessary, since we would then effectively add a log entry for every single message not intended for the `multiListener`, where it's already sort of implied that the `multiListener` doesn't care about it.

Reply:
Right. I was under the wrong impression that the topic check was operated by the `Publisher`, but this is not the case. Indeed the TODO can be removed